src-tauri/src/app/app_data.rs

```diff
@@ -36,10 +36,12 @@ pub(crate) struct LauncherOptions {
     pub custom_data_path: String,
     #[serde(rename = "showNightlyBuilds")]
     pub show_nightly_builds: bool,
-    #[serde(rename = "memoryPercentage")]
-    pub memory_percentage: i32,
     #[serde(rename = "customJavaPath", default)]
     pub custom_java_path: String,
+    #[serde(rename = "systemMemory")]
+    pub system_memory: i64,
```

Comment:
I think you should consider to re-add `mem_percentage` instead of storing the system memory inside the config.